### PR TITLE
Python3.7

### PR DESCRIPTION
**Pull request by CJ-Wright**
(I opened a new PR so I didn't force push to @justcalamari's branch)

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@justcalamari @mariusvniekerk @scopatz @isuruf ready for review!

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -121,6 +121,43 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
+        The list of migrators to run.
+
+    """
+
+    pygraph = copy.deepcopy(gx)
+    r_graph = copy.deepcopy(gx)
+    compiler_graph = copy.deepcopy(gx)
+    openblas_graph = copy.deepcopy(gx)
+    for node, attrs in gx.node.items():
+        if (('python' not in (attrs.get('meta_yaml', {}).get('requirements', {}).get('host', []) or []))
```
Comment (isuruf) [inline conda_forge_tick/auto_tick.xsh]: Can you store `attrs.get('meta_yaml', {}).get('requirements', {})` in a variable? It's hard to read here.

Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.xsh]: Sure

File: conda_forge_tick/auto_tick.xsh [hunk 2]
```diff
@@ -121,6 +121,43 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
+        if 'openblas' not in (attrs.get('meta_yaml', {}).get('requirements',
+                                                          {}).get('host', []) or []):
+            openblas_graph.remove_node(node)
+    total_graph = nx.DiGraph()
+    for g in [pygraph, compiler_graph, r_graph, openblas_graph]:
+        total_graph = nx.compose(total_graph, g)
+    migrators.append(
+        Rebuild(graph=total_graph,
+                pr_limit=1,
+                name='Python 3.7, GCC 7, R 3.5.1, openBLAS 0.3.2'))
```
Comment (isuruf) [inline conda_forge_tick/auto_tick.xsh]: Are you going to build R 3.5.1 packages with both compilers?

Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.xsh]: I think so, @mariusvniekerk @scopatz @ocefpaf?

Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.xsh]: Can that be done in the migrator or does that need to be in the pinnning?

**Comment by CJ-Wright** [comment]
This PR is blocked by:
- [ ] https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/87
and maybe
- [ ] https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/107

File: conda_forge_tick/auto_tick.xsh [hunk 3]
```diff
@@ -121,6 +121,45 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
     return migrate_return, pr_json
 
 
+def build_host(req):
```
Comment (scopatz) [inline conda_forge_tick/auto_tick.xsh]: Public functions should have docstrings

File: conda_forge_tick/migrators.xsh [hunk 4]
```diff
@@ -600,6 +601,37 @@ class Rebuild(Migrator):
         return n
 
 
+class CompilerRebuild(Rebuild):
+    bump_number = 1000
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        with indir(recipe_dir):
+            with indir('..'):
```
Comment (scopatz) [inline conda_forge_tick/migrators.xsh]: It seems like you could save an indentation level by doing `with indir(recipe_dir), indir('..'):` or better ye, just `with indir(recipe_dir + '/..'):`

File: conda_forge_tick/migrators.xsh [hunk 5]
```diff
@@ -600,6 +601,37 @@ class Rebuild(Migrator):
...
+                from ruamel.yaml import safe_load, safe_dump
```
Comment (scopatz) [inline conda_forge_tick/migrators.xsh]: Is there a reason not to have these imports at module level scope? We are already importing conda after all...

**Comment by scopatz** [comment]
A few minor suggestions, but looks good overall!

**Comment by CJ-Wright** [comment]
We can skip Windows for pure compiler packages.

**Comment by CJ-Wright** [comment]
Might wait for new GCC (8) too

**Comment by CJ-Wright** [comment]
I'm going to provisionally merge this and see how it goes. There may be some turbulence for a few days, since there are always bugs to work out in these things and I'll just push to master if there aren't major changes that need to go in. By the same token I may need to shut down this migrator if things seem like they will need lots of help.

**Pull request merged**